### PR TITLE
refactor: Put tests in same file that defines the Structure under test

### PR DESCRIPTION
**Pull request by alamb**
The struct for `TimestampRange` was moved into a module but its tests did not go along for the ride. Put the tests in the same file as their definition

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: query/src/predicate.rs [hunk 1]
```diff
@@ -175,3 +175,32 @@ impl PredicateBuilder {
...
+        assert!(!range.contains_opt(Some(99)));
+        assert!(range.contains_opt(Some(100)));
+        assert!(range.contains_opt(Some(101)));
+        assert!(range.contains_opt(Some(199)));
+        assert!(!range.contains_opt(Some(200)));
+        assert!(!range.contains_opt(Some(201)));
+
+        assert!(!range.contains_opt(None));
+    }
+}
```
Comment (e-dard) [inline query/src/predicate.rs:206]: An alternative suggestion:

```suggestion
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_timestamp_range_contains() {
        let range = TimestampRange::new(100, 200);
        assert!([100, 101, 199].iter().all(|&ts| range.contains(ts)));
        assert!([99, 200, 201].iter().all(|&ts| !range.contains(ts)));
    }

    #[test]
    fn test_timestamp_range_contains_opt() {
        let range = TimestampRange::new(100, 200);
        assert!([100, 101, 199]
            .iter()
            .all(|&ts| range.contains_opt(Some(ts))));
        assert!([99, 200, 201]
            .iter()
            .all(|&ts| !range.contains_opt(Some(ts))));

        assert!(!range.contains_opt(None));
    }
}
```

**Review by e-dard — APPROVED** [review]
LGTM 👍

**Comment by alamb** [comment]
I didn't see the suggestion before clicking merge -- I will make the change in a subsequent PR

File: query/src/predicate.rs (hunk 1 above)
Comment (alamb) [reply query/src/predicate.rs:206]: While starting to implement this change with a fresh pair of eyes, I conclude that I personally prefer the more verbose form of the test. 

My rational:  if this test ever fails, the more verbose form will tell you exactly which value failed. 

With the more compact formulation, for example, if this fails:
```
        assert!([100, 101, 199].iter().all(|&ts| range.contains(ts)));
```

All you will know is that it was either 100, 101, or 199.

Thus I plan to not implement this suggestion at this time.

**Pull request merged**